File: ProjecIntegration.Api/projectintegrationFront/src/app/class/salledetheatre.ts
import { Complexe } from "./complexe";
import { Representation } from "./representation";

export class SalledeTheatre { Name: string;
  PlaceMax: number;
  PlaceCurrent: number;
  representations: Representation[];
  Complexe: Complexe;
  ComplexeId: number;

  constructor(
    Name: string,
    PlaceMax: number,
    PlaceCurrent: number,
    representations: Representation[],
    Complexe: Complexe,
    ComplexeId: number
  ) {
    this.Name = Name;
    this.PlaceMax = PlaceMax;
    this.PlaceCurrent = PlaceCurrent;
    this.representations = representations;
    this.Complexe = Complexe;
    this.ComplexeId = ComplexeId;
  }
}

File: ProjecIntegration.Api/Domain/Entity/TheatherEntity/SalleDeTheatre.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Entity.TheatherEntity
{
    public class SalleDeTheatre : BaseEntity
    {

        public string Name { get; set; }=string.Empty;  
        public int PlaceMax { get; set; }
        public int ComplexeId { get; set; }
        public Complexe? Complexe { get; set; }
        public List<Representation>? Representations { get; set; }
 
        public List<Siege> sieges { get; set; }
    }
}
